### DISCUSSION TITLE
How can I configure WiFi SoftAP on Zephyr RTOS with ESP32S3 without using shell commands.

### DISCUSSION THREAD

**Discussion by SanjayEIC** [zephyrproject-rtos/zephyr#75200] [General]
I am using Zephyr RTOS with ESP32S3, and I need to configure WiFi SoftAP without using shell commands. Currently, in the Zephyr sample, it is done with shell commands, but I want to do it without the shell.

**Comment by ndrs-pst** [comment]
To configure Wi-Fi SoftAP without using shell commands, you can extract the code responsible for setting up the Wi-Fi from the `wifi_shell.c` file and run it within your own thread.

e.g. from `cmd_wifi_mode` function you may extract the following code

```
struct net_if *iface;
struct wifi_mode_info mode_info = {0};

mode_info.oper = WIFI_MGMT_SET;
mode_info.mode = WIFI_SOFTAP_MODE;
iface = net_if_get_first_wifi();
mode_info.if_index = net_if_get_by_iface(iface);

ret = net_mgmt(NET_REQUEST_WIFI_MODE, iface, &mode_info, sizeof(mode_info));
...
```

**Comment by SanjayEIC** [comment]
Hi @ndrs-pst 
sorry for late reply  
I tried you suggested, but I am still facing the problem.
<img width="719" alt="SoftAP" src="https://github.com/zephyrproject-rtos/zephyr/assets/126941623/56b24e61-ce18-4fef-a10b-8350bd869fb1">

**Comment by SanjayEIC** [comment]
I am trying another example also, with some documents and other online websites, so Soft AP is enabled and it is broadcasting, but when I use a static IP address and enter the router IP to connect, the connection is established. However, when I ping the IP, it gives an error.

<img width="713" alt="wifi_AP" src="https://github.com/zephyrproject-rtos/zephyr/assets/126941623/8262cd06-a12e-4f9c-a027-10e746ece2af">

this is configuration :- 
<img width="292" alt="config" src="https://github.com/zephyrproject-rtos/zephyr/assets/126941623/a737200c-89ae-4d78-91e6-3434a0f08199">

socket code :- 
<img width="629" alt="TCP_IP" src="https://github.com/zephyrproject-rtos/zephyr/assets/126941623/478383f0-734c-4bbd-a6ca-8471de366dba">

I follow the steps :- WiFi scan --- > enter password , set static IP and all details , after that i am using for communication , using "serial WiFi terminal app" but getting error

![error](https://github.com/zephyrproject-rtos/zephyr/assets/126941623/9d5c6e83-0606-4ae1-86a0-cd1922b1f8ff)
   

Can you suggest if I have made a mistake in enabling Wi-Fi or Socket  if any code is missing?

**Reply by ndrs-pst** [reply]
I may ping this guy @krish2718 😉

**Reply by krish2718** [reply]
> However, when I ping the IP, it gives an error.

To confirm, you have assigned 192.0.2.8 static IP to the STA and ping to SAP (192.0.2.1) fails? Can you please also check the network masks?

Please also share output of `net iface`  and also `wifi ap stations` and try pinging from SAP to STA (you need SHELL  and NET SHELL enabled for this).

**Reply by SanjayEIC** [reply]
1) To confirm, you have assigned 192.0.2.8 static IP to the STA and ping to SAP (192.0.2.1) fails?
Yes it fails 

configuration :- In mobile (STA) 
![configuation](https://github.com/zephyrproject-rtos/zephyr/assets/126941623/e82a378d-87b0-49b4-8c38-5aeb8458e443)

2) this is the iface return [00:00:00.381,000] <inf> main: Interface Pointer: **0x3fcb08c0**

3) I ping using shell command loop back is working but i ping to STA getting error 
<img width="398" alt="ping" src="https://github.com/zephyrproject-rtos/zephyr/assets/126941623/d511375f-3049-473c-8348-ab5422190791">

**Comment by krish2718** [comment]
> this is the iface return [00:00:00.381,000] main: Interface Pointer: 0x3fcb08c0

This is not the shell output, if you cannot use shell, then you can replicate code from https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/net/lib/shell/iface.c#L634 and https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/net/l2/wifi/wifi_shell.c#L1307, basically I would

1. the IP,mask combinations
2. interface status
3. Wi-Fi STA status in SAP
4. Check ARP resolution in both ends: `net arp` in Zephyr and `ip n show` in Linux
5. Confirm routing in STA, `ip r` should do.
6. `net stats` and `wifi statistics` (you really should enable SHELL + NET SHELL for debugging only, else you end up copying loads of code).
